generate.go

```diff
@@ -232,172 +236,58 @@ func modify(spec *rspec.Spec, context *cli.Context) error {
-								"SCMP_CMP_GT|SCMP_CMP_MASKED_EQ")
-						}
-						op := rspec.Operator(args[3])
-						Arg := rspec.Arg{
-							Index:    uint(index),
-							Value:    uint64(value),
-							ValueTwo: uint64(value2),
-							Op:       op,
-						}
-						Args = append(Args, Arg)
-					} else {
-						return fmt.Errorf("seccomp-sysctl args error: %s", argsstru)
-					}
-				}
-			}
-
-			syscallstruct := rspec.Syscall{
-				Name:   name,
-				Action: action,
-				Args:   Args,
-			}
-			secc.Syscalls = append(secc.Syscalls, syscallstruct)
-		} else {
-			return fmt.Errorf("seccomp sysctl must consist of 3 parameters")
-		}
-	}
-
-	return nil
-}
-
 func addSeccomp(spec *rspec.Spec, context *cli.Context) error {
 	var secc rspec.Seccomp
-	sd := context.String("seccomp-default")
-	sa := context.StringSlice("seccomp-arch")
-	ss := context.StringSlice("seccomp-syscalls")
 
-	if sd == "" && len(sa) == 0 && len(ss) == 0 {
-		return nil
-	}
+	seccompDefault := context.String("seccomp-default")
+	seccompArch := context.String("seccomp-arch")
+	seccompKill := context.String("seccomp-kill")
+	seccompTrace := context.String("seccomp-trace")
+	seccompErrno := context.String("seccomp-errno")
+	seccompTrap := context.String("seccomp-trap")
+	seccompAllow := context.String("seccomp-allow")
 
 	// Set the DefaultAction of seccomp
-	err := addSeccompDefault(spec, sd, &secc)
+	if seccompDefault == "" {
+		seccompDefault = "allow"
+	}
+	err := seccomp.ParseDefaultAction(seccompDefault, &secc)
 	if err != nil {
 		return err
 	}
 
 	// Add the additional architectures permitted to be used for system calls
-	err = addSeccompArch(spec, sa, &secc)
+	if seccompArch == "" {
+		seccompArch = "amd64,x86,x32"
+	}
+	err = seccomp.ParseArchitectureFlag(seccompArch, &secc)
 	if err != nil {
 		return err
 	}
 
-	// Set syscall restrict in Seccomp
-	// The format of input syscall string is Name:Action:Args[1],Args[2],...,Args[n]
-	// The format of Args string is Index/Value/ValueTwo/Operator,and is parsed by function parseArgs()
-	err = addSeccompSyscall(spec, ss, &secc)
+	err = seccomp.ParseSyscallFlag("kill", seccompKill, &secc)
 	if err != nil {
 		return err
 	}
 
-	for _, a := range context.StringSlice("seccomp-allow") {
-		sysCall := rspec.Syscall{
-			Name:   a,
-			Action: "SCMP_ACT_ALLOW",
-		}
-		secc.Syscalls = append(secc.Syscalls, sysCall)
+	err = seccomp.ParseSyscallFlag("trace", seccompTrace, &secc)
+	if err != nil {
+		return err
 	}
 
-	for _, a := range context.StringSlice("seccomp-errno") {
-		sysCall := rspec.Syscall{
-			Name:   a,
-			Action: "SCMP_ACT_ERRNO",
-		}
-		secc.Syscalls = append(secc.Syscalls, sysCall)
+	err = seccomp.ParseSyscallFlag("errno", seccompErrno, &secc)
+	if err != nil {
+		return err
+	}
+
+	err = seccomp.ParseSyscallFlag("trap", seccompTrap, &secc)
+	if err != nil {
+		return err
+	}
+
+	err = seccomp.ParseSyscallFlag("allow", seccompAllow, &secc)
+	if err != nil {
+		return err
 	}
 
 	spec.Linux.Seccomp = &secc
```

Comment:
The function `parseArgs()` follows this one is useless with your commit, so you can remove that.